#! /bin/sh /usr/share/dpatch/dpatch-run
## 12-options by  <adamm@zombino.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes inconsistant behaviour of select tag.

@DPATCH@
--- a/actionpack/lib/action_view/helpers/form_options_helper.rb	2006-08-10 13:39:10.000000000 -0500
+++ b/actionpack/lib/action_view/helpers/form_options_helper.rb	2006-08-18 15:44:21.000000000 -0500
@@ -298,14 +298,16 @@
 
       def to_select_tag(choices, options, html_options)
         html_options = html_options.stringify_keys
-        add_default_name_and_id(html_options)
+        options = options.stringify_keys
+        add_default_name_and_id(options)
         selected_value = options.has_key?(:selected) ? options[:selected] : value
         content_tag("select", add_options(options_for_select(choices, selected_value), options, value), html_options)
       end
 
       def to_collection_select_tag(collection, value_method, text_method, options, html_options)
         html_options = html_options.stringify_keys
-        add_default_name_and_id(html_options)
+        options = options.stringify_keys
+        add_default_name_and_id(options)
         content_tag(
           "select", add_options(options_from_collection_for_select(collection, value_method, text_method, value), options, value), html_options
         )
@@ -313,13 +315,15 @@
 
       def to_country_select_tag(priority_countries, options, html_options)
         html_options = html_options.stringify_keys
-        add_default_name_and_id(html_options)
+        options = options.stringify_keys
+        add_default_name_and_id(options)
         content_tag("select", add_options(country_options_for_select(value, priority_countries), options, value), html_options)
       end
 
       def to_time_zone_select_tag(priority_zones, options, html_options)
         html_options = html_options.stringify_keys
-        add_default_name_and_id(html_options)
+        options = options.stringify_keys
+        add_default_name_and_id(options)
         content_tag("select",
           add_options(
             time_zone_options_for_select(value, priority_zones, options[:model] || TimeZone),
